plsync/planetlab/model.py

```diff
@@ -340,11 +354,19 @@ def __init__(self, **kwargs):
         super(PCU, self).__init__(**kwargs)
 
     def hostname(self):
-        """ generate the hostname for this DRAC based on index & site name """
-        return 'mlab%dd.%s.%s' % (self['index'], self['name'], MLAB_ORG_DOMAIN)
+        """Returns the hostname for this DRAC."""
+        return '.'.join((self.recordname(), MLAB_ORG_DOMAIN))
```

Comment:
I agree it makes for a bunch of parentheses. The argument to join needs to be an iterable type, for example a list or tuple. Here it's a tuple.